src/libraries/System.Private.CoreLib/src/System/Guid.cs

```diff
@@ -444,34 +470,39 @@ public static bool TryParseExact(ReadOnlySpan<char> input, [StringSyntax(StringS
                 return false;
             }
         }
-
-        private static bool TryParseGuid(ReadOnlySpan<char> guidString, ref GuidResult result)
+        private static bool TryParseGuid<TChar>(ReadOnlySpan<TChar> guidString, ref GuidResult result) where TChar : unmanaged, IUtfChar<TChar>
         {
-            guidString = guidString.Trim(); // Remove whitespace from beginning and end
+            guidString = Number.SpanTrim(guidString); // Remove whitespace from beginning and end
 
             if (guidString.Length < 32) // Minimal length we can parse ('N' format)
             {
                 result.SetFailure(ParseFailure.Format_GuidUnrecognized);
                 return false;
             }
-
-            return (guidString[0]) switch
+            if (guidString[0] == TChar.CastFrom('('))
```

Comment:
Does this need to change structure, or could it retain its previous structure by casting guidString[0] to char and continuing to switch on that?